### PR TITLE
fix(helm): fix two panics in 'helm search'

### LINKED ISSUE #778: helm search raises a runtime error unless helm has been initialized
After installing helm from source, if I execute `helm search redis` I get:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x40dfd0]

goroutine 1 [running]:
panic(0x1905e80, 0xc82000e120)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
main.searchCacheForPattern.func1(0xc8203df680, 0x26, 0x0, 0x0, 0x7f53d2f6f2d0, 0xc8203df800, 0x0, 0x0)
	/home/migmartri/work/gocode/src/github.com/kubernetes/helm/cmd/helm/search.go:63 +0x40
```

Problem that seems to be related to the fact that I have not initialized helm yet.

So executing `helm init` fixed the issue.

```
$ helm search redis
kubernetes-charts/redis-2.0.0.tgz
```

I have noticed that for some other commands there is a guard statement that tells you that the file does not exist, for example:

```
$helm fetch foo
Error: open /home/migmartri/.helm/repository/repositories.yaml: no such file or directory
```

We might want to implement the same approach for the search command. Ideally with a message a little bit more actionable that says something about executing helm init first (although this could be moved to another issue)

What do you think?

### PR DESCRIPTION
**Pull request by technosophos**
Closes #778

### CODE REVIEW

File: cmd/helm/search.go [hunk 1]
```diff
@@ -58,6 +61,10 @@ func searchChartRefsForPattern(search string, chartRefs map[string]*repo.ChartRe
 
 func searchCacheForPattern(dir string, search string) ([]string, error) {
 	fileList := []string{}
+	if _, err := os.Stat(dir); err != nil {
+		return fileList, errors.New("No cache exists. Did you run 'helm init'?")
```
Comment (adamreese) [inline cmd/helm/search.go]: should we generate the cache file here?

Comment (technosophos) [inline cmd/helm/search.go]: I think that we'd have to run the entire `helm init` process, which doesn't seem like a good idea to do here. Thoughts?

Comment (vaikas-google) [inline cmd/helm/search.go]: Another thing we could do is have a method that checks if 'helm init' has been ran and just call that from the commands before proceeding? That way we don't have to handle the special cases in these commands.

Comment (technosophos) [inline cmd/helm/search.go]: Good idea. I'm going to try that.

**Comment by technosophos** [comment]
Okay, this new refactor provides a generic solution that other commands can use.

**Pull request merged**